File: 1009 - Complement of Base 10 Integer.js
/**
 * @param {number} n
 * @return {number}
 */
var bitwiseComplement = function (n) {
    let a = n.toString(2);
    let len = a.length;
    let res = 0;
    for (let i = 0; i < len; i++) {
        if (a[len - 1 - i] == '0') {
            res += Math.pow(2, i);
        }
    }
    return res;
};

File: 1009 - Complement of Base 10 Integer.cpp
#include <cmath>

class Solution {
public:
    int bitwiseComplement(int n) {
        int res = 0;
        if (n == 0) return 1;
        int p = 0;
        while (n > 0) {
            if (n % 2 == 0) {
                res += pow(2, p);
            }
            n /= 2;
            p++;
        }
        return res;
        
    }
};